tests/integration/ha_tests/test_ha.py

```diff
@@ -459,3 +476,100 @@ async def test_restart_db_process(ops_test, continuous_writes):
+
+    # verify new writes are continuing by counting the number of writes before and after a 5 second
+    # wait
+    writes = await count_writes(ops_test)
+    time.sleep(5)
+    more_writes = await count_writes(ops_test)
+    assert more_writes > writes, "writes not continuing to DB"
+
+    # verify presence of primary, replica set member configuration, and number of primaries
+    await verify_replica_set_configuration(ops_test)
+
+    # verify that no writes to the db were missed
+    total_expected_writes = await stop_continous_writes(ops_test)
+    actual_writes = await count_writes(ops_test)
+    # verify that no writes were missed. Cluster crash can occasionally lead to a single
+    # duplicate write, hence the `or`.
+    assert (
+        actual_writes == total_expected_writes["number"]
+        or actual_writes == 1 + total_expected_writes["number"]
+    ), "db writes missing."
+
+
+async def test_full_cluster_restart(ops_test: OpsTest, continuous_writes, reset_restart_delay):
+    app = await app_name(ops_test)
+
+    # update all units to have a new RESTART_DELAY,  Modifying the Restart delay to 3 minutes
+    # should ensure enough time for all replicas to be down at the same time.
+    for unit in ops_test.model.applications[app].units:
+        await update_restart_delay(ops_test, unit, RESTART_DELAY)
+
+    # kill all units "simulatenously"
+    await asyncio.gather(
+        *[
+            kill_unit_process(ops_test, unit.name, kill_code="SIGTERM")
+            for unit in ops_test.model.applications[app].units
+        ]
+    )
+
+    # This test serves to verify behavior when all replicas are down at the same time that when
+    # they come back online they operate as expected. This check verfies that we meet the criterea
+    # of all replicas being down at the same time.
+    assert await all_db_processes_down(ops_test), "Not all units down at the same time."
+
+    # sleep for twice the median election time and the restart delay
+    time.sleep(MEDIAN_REELECTION_TIME * 2 + RESTART_DELAY)
+
+    # verify all units are up and running
+    for unit in ops_test.model.applications[app].units:
+        assert await mongod_ready(
+            ops_test, unit.public_address
+        ), f"unit {unit.name} not restarted after cluster crash."
+
+    # verify new writes are continuing by counting the number of writes before and after a 5 second
+    # wait
+    writes = await count_writes(ops_test)
+    time.sleep(5)
+    more_writes = await count_writes(ops_test)
+    assert more_writes > writes, "writes not continuing to DB"
+
+    # verify presence of primary, replica set member configuration, and number of primaries
+    await verify_replica_set_configuration(ops_test)
+
+    # verify that no writes to the db were missed
+    total_expected_writes = await stop_continous_writes(ops_test)
+    actual_writes = await count_writes(ops_test)
+    assert total_expected_writes["number"] == actual_writes, "writes to the db were missed."
```

Comment:
Out of curiosity, does Mongo allow for duplicate writes as a KV store?
I suppose it might if you `insert()`, but I wonder if normally people `update()`?

Reply:
@marceloneppel hmm I don't think so but not 100% sure

@marcoppenheimer  good point, Mykola and I think that is a good fix for now :)